packages/block-editor/src/components/block-list/block.js

```diff
@@ -402,7 +403,8 @@ function BlockListBlock( {
 		isSelected &&
 		! showEmptyBlockSideInserter &&
 		! isPartOfMultiSelection &&
-		! isTypingWithinBlock;
+		! isTypingWithinBlock &&
+		! horizontalMover;
```

Comment:
It seems that shouldRenderMovers is going to be false even if it should show the horizontal movers, shouldRenderMovers should not depend if the mover is horizontal or vertical otherwise the logic is complex and it will be easy to introduce bugs in the future.

Reply:
removed that with the new rendering position